### PR TITLE
Add Sass tests

### PR DESCRIPTION
**Pull request by drwpow**
## Changes

Tests that #1813 doesn’t regress (issue fixed in previous style work; this just adds test to verify)

No code changes, only Sass tests. Can be merged whenever.

Tests on all frameworks:

- ✅ Astro (scss/sass)
- ✅ `<link>` tag (scss/sass)
- ✅ JSX (scss/sass)
- ✅ Vue (scss/sass)
- ❌ Svelte (there seems to be a problem with Svelte + Sass; test skipped for now)

Renames `astro-styles-ssr.test.js` to `styles-ssr.test.js` because we’re testing a lot more than Astro right now!

## Testing

Only tests

## Docs

No docs needed

### CODE REVIEW

File: packages/astro/test/fixtures/styles-ssr/src/components/ReactModules.jsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import Styles from './ReactModules.module.css';
 
 function ReactModules() {
   return (
-    <h1 id="react-modules" className={Styles.title}>
+    <h1 id="react-module-css" className={Styles.title}>
```
Comment (drwpow) [inline packages/astro/test/fixtures/styles-ssr/src/components/ReactModules.jsx:6]: All changes in this test file are arbitrary; just things like making the classnames between tests easier to distinguish from one another.

Barring insignificant changes, no tests were removed, but 12 new ones were added

**Review by jonathantneal — APPROVED** [review]

**Comment by drwpow** [comment]
Not sure why the tests are timing out on CI. They’re passing locally. Will keep digging into it but also can revisit later.

File: packages/astro/package.json [hunk 2]
```diff
@@ -90,7 +90,7 @@
     "remark-slug": "^7.0.0",
     "resolve": "^1.20.0",
     "rollup": "^2.57.0",
-    "sass": "^1.43.3",
+    "sass": "^1.43.4",
```
Comment (natemoo-re) [inline packages/astro/package.json:93]: Needs a changeset `patch` for dependency bump.

File: packages/astro/test/0-css.test.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+      const scopedClass = classes.find((name) => /^s-[A-Za-z0-9-]+/.test(name));
+
+      // 1. check HTML
+      expect(el.attr('class')).to.include('svelte-css');
+
+      // 2. check CSS
+      expect(bundledCSS).to.match(new RegExp(`.svelte-css.${scopedClass}[^{]*{font-family:"Comic Sans MS"`));
+    });
+
+    // TODO: fix Sass in Svelte
```
Comment (natemoo-re) [inline packages/astro/test/0-css.test.js:241]: Do we have an issue to track this? I want to get in the habit of creating issues for `TODO` comments.

**Review by natemoo-re — APPROVED** [review]
Looks great! Non-blocking comment on `TODO` comments. ☺️

File: packages/astro/test/0-css.test.js (hunk 3 above)
Comment (drwpow) [reply packages/astro/test/0-css.test.js:241]: We do! https://github.com/snowpackjs/astro/issues/1976

**Comment by drwpow** [comment]
Update: tried running tests in non-parallel mode, and it’s _waaaay_ slower. I want to say this isn’t _flaky_; just a side-effect of `--parallel` mode which I don’t want to give up just yet.

If I catch wind of these tests failing I’ll do something else, but I think just running certain tests first is the best solution (unless memory fixes improve this!)

**Pull request merged**